.github/workflows/rockon_validator.yaml

```
@@ -0,0 +1,57 @@
+# Use rockon-validator repo to validate style and completeness of new Rockon submissions
+# Thanks to @StephenBrown2 for donating original validation repository to the Rockstor project.
+
+name: rockon-validator
+
+on:
+  # Triggers the workflow on push or pull request events but only for the "master" branch
+  push:
+    branches: [ "master" ]
+    paths:
+    - '*.json'
+    
+  pull_request:
+    branches: [ "master" ]
+    paths:
+    - '*.json'
+    
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  # This workflow contains a single job called "validate"
+  validate:
+    name: rockon validator
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+
+      # install the latest version of Rockon validator
+      - name: Setup latest Rockon validator
+        run: |
+         go install github.com/rockstor/rockon-validator@latest 
+
+      - name: Get all changed json files
+        id: changed-json-files
+        uses: tj-actions/changed-files@v45
+        with:
+          # json files
+          files: |
+            **.json
+
+# show differentials based on ideal Rockon schema and whether it's in the root.json file
+      - name: Process all changed json files
+        if: steps.changed-json-files.outputs.any_changed == 'true'
+        env:
+          ALL_CHANGED_FILES: ${{ steps.changed-json-files.outputs.all_changed_files }}
+        run: |
+          for jsonfile in ${ALL_CHANGED_FILES}; do
+            rockon-validator -d $jsonfile
+          done
```

If this is just meant to be informative, and not fail if the spec isn't met, this is fine. If you want to fail the check if there is a diff to be changed, you can add the `-c, --check` flag.

Additionally, the tool can parse multiple files at once, so the step can be simplified to the following:

```suggestion
        run: rockon-validator -c -d ${{ steps.changed-json-files.outputs.all_changed_files }}
```

---

@StephenBrown2 I experimented with the `-c` flag, but figured that to start with it would be easier to look at the output as a guidance to what would look different. thanks for the suggestion, I just couldn't make it work originally with multiple files (as I was doing it wrong to begin with ... ) hence I arrived at the above.
qq since I have your attention already: can I also add the root file check into the same command line?

---

Sorry my attention is not very good these days. ;-)

The root file check is always done, passing `--root <path>` will just tell the tool to use a different `root.json` file. Let me know if there's anything I should change in https://github.com/rockstor/rockon-validator/tree/main?tab=readme-ov-file#rootjson to make the usage clearer.

---

no worries at all. Thanks for the clarification. I wasn't entirely clear on that. I will post another question in a separate thread to you on the docker option that you provided.